Sources/SpeziAccessGuard/CodeViews/CodeView.swift

```diff
@@ -63,11 +66,27 @@ struct CodeView: View {
                 oldKeyBoardType = codeOption.keyBoardType
             }
             .onChange(of: code) {
-                if code.count >= codeOption.maxLength {
-                    Task { @MainActor in
-                        await checkCode()
+                ///Validation is skipped, when code was reset or modified after an error
+                if formerCode == code || code == "" {
+                    return
+                }
+                
+                let validationRes = codeOption.continousValidation(ofCode: code)
+                
+                Task { @MainActor in
+                    do {
+                        try await action(code, validationRes)
+                    } catch {
+                        code = ""
                     }
                 }
+                
+                switch validationRes {
+                    case .failure(_):
```

Comment:
The associated value binding can be omitted:
```suggestion
                    case .failure:
```